adr/0002-reactive-rename.adoc

```diff
@@ -0,0 +1,73 @@
+= Rename Reactive extensions
+
+* Status: proposed
+* Date: 2023-10-17
+
+== Context and Problem Statement
+
+RESTEasy Reactive is the preferred REST stack in Quarkus, however in the almost 3 years since its release, we have seen
+folks of all experience misunderstand its capabilities and being led to believe that it forces reactive programming on them.
+
+With the advent of Virtual Threads in Java 19+, the problem becomes larger since in the minds of users, Reactive and Virtual Threads
+are two different worlds.
+
+The same problem applies to other extensions as well:
+
+* Reactive Messaging
+* Reactive Routes
+* REST Client Reactive
```

Comment:
OIDC Client Reactive, OIDC Token Propagation Reactive and Keycloak Admin Client Reactive are other 3 extensions which have a reactive qualifier, all of them work with REST Client Reactive.